From 372e4cbb8670ac3d0de4b6c65d55de934e508369 Mon Sep 17 00:00:00 2001
From: Antoine Damhet <antoine.damhet@lse.epita.fr>
Date: Thu, 19 Oct 2023 19:51:12 +0200
Subject: [PATCH] Install data files from the Makefile

This commit reverts 74e8aa8b1c81798e2f6c5e122a335ba89ebe89b1 and part of
f2491ba7725a1ae1b47ab0d2d843578a6f04800e.

Signed-off-by: Antoine Damhet <antoine.damhet@lse.epita.fr>
---
 Makefile | 41 +++++++++++++++++++++++++++++++++++------
 setup.py | 35 +----------------------------------
 2 files changed, 36 insertions(+), 40 deletions(-)

diff --git a/Makefile b/Makefile
index 4b39c34a921c..24da1d1bdcc8 100644
--- a/Makefile
+++ b/Makefile
@@ -1,16 +1,45 @@
+PREFIX ?= /usr
 DESTDIR ?= /
+LIBDIR ?= $(PREFIX)/lib
+SHAREDIR = $(PREFIX)/share
+MANDIR ?= $(SHAREDIR)/man
+PYTHON ?= yes
+
+EXTENSION_DIR ?= $(LIBDIR)/password-store/extensions
+
+BASHCOMPDIR ?= $(SHAREDIR)/bash-completion/completions
+ZSHCOMPDIR ?= $(SHAREDIR)/zsh/site-functions
 
 all:
 	@python3 setup.py build
 	@echo "pass-audit was built successfully. You can now install it wit \"make install\""
 
-install:
-	@python3 setup.py install --root="$(DESTDIR)" --optimize=1 --skip-build
-	@echo "pass-audit is installed succesfully"
+base_install:
+	@install -vd "$(DESTDIR)$(EXTENSION_DIR)/" "$(DESTDIR)$(MANDIR)/man1" \
+		     "$(DESTDIR)$(BASHCOMPDIR)" "$(DESTDIR)$(ZSHCOMPDIR)"
+	@install -vm 0755 audit.bash "$(DESTDIR)$(EXTENSION_DIR)/audit.bash"
+	@install -vm 0644 share/man/man1/pass-audit.1 "$(DESTDIR)$(MANDIR)/man1/"
+	@install -vm 0644 share/bash-completion/completions/pass-audit "$(DESTDIR)$(BASHCOMPDIR)/"
+	@install -vm 0644 share/zsh/site-functions/_pass-audit "$(DESTDIR)$(ZSHCOMPDIR)/"
+
+install: base_install
+	@[ "$(PYTHON)" = "yes" ] || exit 0; python3 setup.py install --root="$(DESTDIR)" --optimize=1 --skip-build
+	@echo "pass-audit is installed successfully"
+
+uninstall:
+	@rm -vrf \
+		"$(DESTDIR)$(EXTENSION_DIR)/audit.bash" \
+		"$(DESTDIR)$(MANDIR)/man1/pass-audit.1" \
+		"$(DESTDIR)$(ZSHCOMPDIR)/_pass-audit" \
+		"$(DESTDIR)$(BASHCOMPDIR)/pass-audit"
 
-local:
-	@python3 setup.py install --user --optimize=1
-	@echo "pass-audit is localy installed succesfully."
+XDG_DATA_HOME ?= $(HOME)/.local/share
+PASSWORD_STORE_EXTENSIONS_DIR ?= $(HOME)/.password-store/.extensions
+local: EXTENSION_DIR = $(PASSWORD_STORE_EXTENSIONS_DIR)
+local: SHAREDIR = $(XDG_DATA_HOME)
+local: base_install
+	@[ "$(PYTHON)" = "yes" ] || exit 0; python3 setup.py install --user --optimize=1
+	@echo "pass-audit is locally installed successfully."
 	@echo "Remember to set PASSWORD_STORE_ENABLE_EXTENSIONS to 'true' for the extension to be enabled."
 
 tests:
diff --git a/setup.py b/setup.py
index 1f0a58b92552..2eb76a77466f 100644
--- a/setup.py
+++ b/setup.py
@@ -2,39 +2,6 @@
 # pass audit - Password Store Extension (https://www.passwordstore.org/)
 # Copyright (C) 2017-2022 Alexandre PUJOL <alexandre@pujol.io>.
 
-import os
-import sys
-from pathlib import Path
-
 from setuptools import setup
 
-share = Path(sys.prefix, 'share')
-base = '/usr'
-if os.uname().sysname == 'Darwin':
-    base = '/usr/local'
-lib = Path(base, 'lib', 'password-store', 'extensions')
-
-if '--user' in sys.argv:
-    if 'PASSWORD_STORE_EXTENSIONS_DIR' in os.environ:
-        lib = Path(os.environ['PASSWORD_STORE_EXTENSIONS_DIR'])
-    else:
-        lib = Path.home() / '.password-store' / '.extensions'
-    if 'XDG_DATA_HOME' in os.environ:
-        share = Path(os.environ['XDG_DATA_HOME'])
-    else:
-        share = Path.home() / '.local' / 'share'
-
-setup(
-    data_files=[
-        (str(share / 'man' / 'man1'), [
-            'share/man/man1/pass-audit.1'
-        ]),
-        (str(share / 'bash-completion' / 'completions'), [
-            'share/bash-completion/completions/pass-audit'
-        ]),
-        (str(share / 'zsh' / 'site-functions'), [
-            'share/zsh/site-functions/_pass-audit'
-        ]),
-        (str(lib), ["audit.bash"]),
-    ],
-)
+setup()
-- 
2.41.0

